### ISSUE TITLE
Can no longer instantiate subclasses

### ISSUE THREAD

**Issue by mcanders** [godotengine/godot#3871]
**Operating system or device:**
Arch Linux

**Issue description** (what happened, and what was expected):
I get an "Invalid call. Nonexistent function 'new' in base 'GDScript'." error when I attempt to instantiate a subclass defined within a class file.  This has worked in the past and broke within the last week in master.

Example:
```python
var Bar = preload("Bar.gd")

class Foo:
	var x = 0
	func new(y):
		x = y


func _ready():
	# This works
	var bar = Bar.new(1)

	# This does not work
	var foo = Foo.new(1)

```

**Steps to reproduce:**
I provided a simple sample project.

**Link to minimal example project** (optional but very welcome):
[SubclassTest.zip](https://github.com/godotengine/godot/files/150203/SubclassTest.zip)

**Comment by trollworkout** [comment]
i'm pretty sure you use it like 
var bar  = Bar()
far foo= Foo()

**Comment by mcanders** [comment]
That syntax is only used for the internal types (like Points and Vectors).  .new() and subclasses are well documented here: http://docs.godotengine.org/en/latest/reference/gdscript.html?highlight=gdscript

**Comment by trollworkout** [comment]
I see the subclass mentioned in the docs. I wonder if that's outdated and is no longer possible. We will wait for one of the devs to respond.

**Comment by akien-mga** [comment]
I believe you need to name the constructor ``_init`` and not ``new``.

**Comment by mcanders** [comment]
Apologies.  The example is wrong (I hastily put it together), but the issue does actually exist.  Switching the "new" out for "_init" DOES result in the error I provided.

**Comment by mcanders** [comment]
But weirdly results in the same error :smile:

**Comment by mcanders** [comment]
Aaaand wrong again.  The errors are different.

Error using "_init" : "Invalid call. Nonexistent function 'new' in base 'GDScript'."
Error using "new": Cannot access member without instance

**Comment by akien-mga** [comment]
For me this script works just fine:
```
extends Node

class Foo:
	var x = 0
	func _init(y):
		x = y

func _ready():
	var foo = Foo.new(1)
	print(foo.x)
```
It prints "1" as expected.

Testing with 2.0-stable on Linux.

**Comment by mcanders** [comment]
2.0-stable does not have this problem for me either.  It is a problem currently in master.

**Comment by akien-mga** [comment]
Right, I just built master HEAD from today and can reproduce the bug. I did not get the bug with a build from sometime yesterday or the day before I think, so it appears to be quite recent.

**Comment by bojidar-bg** [comment]
:+1:, just occurred to me yesterday, as I was testing my Tween docs PR.

**Comment by reduz** [comment]
ah I found the issue, tried to fix a crash but introduced this bug. Will
push a fix in a bit

**Comment by bojidar-bg** [comment]
@reduz gah, I was bisecting whole morning :smile:. Final bisect output:
````
6fc1c3a4d1cf0c865f7dfdb1221ef07a5d25f305 is the first bad commit
commit 6fc1c3a4d1cf0c865f7dfdb1221ef07a5d25f305
Author: Juan Linietsky <reduzio@gmail.com>
Date:   Sat Feb 27 23:10:44 2016 -0300

    Completed the support for plugins! It is not possible to add plugins.
    Not all APIs are provided yet, please request whathever you are missing.
    Some example plugins are provided in demos/plugins. Just copy them to a folder in your project named addons/ and then enable them from the project settings.
    Have fun!

:040000 040000 a30f13a45204060cf877c1f4324c2d9fcbf1b1b9 d0040b6cd5881a05612104c1fca7d2b5aad05881 M      core
:040000 040000 29b8fda657182588d8da091b396c0fdb19523483 95c5c69a1ed702f5f2e905a872e42bc1715b8ace M      demos
:040000 040000 b2f8733324903e9729e1c61b0333dcbb4e992b0f 342633be6ef42daf6ecf0937533371e231b9fe0c M      doc
:040000 040000 70517af8bacddc2b847d50faa6f3bd7b42ca320f a7c7442c55b3a64083837ac7f1f0da80ab454088 M      modules
:040000 040000 24c218d746afea166d610c1b550913d81a44b2f0 9a8bb1729c19091df98e26cd78de7a1b0ca5247e M      tools
````

**Comment by reduz** [comment]
pushed a fix, please try it

**Comment by bojidar-bg** [comment]
@reduz ~ I'm currently compiling (who know why it takes so long this time), and I would write when it finishes.

**Comment by bojidar-bg** [comment]
@reduz ~ Ok, it works! I guess you can close it now.

**Comment by akien-mga** [comment]
Fixed by 210d332.
